tests/core/test_test.py

```diff
@@ -893,6 +893,94 @@ def test_test_generation_with_array(tmp_path: Path) -> None:
         "query": [{"array_col": [["value1"], ["value2", "value3"]]}]
     }
 
+    # Array of maps
+    input_queries = {
+        "sqlmesh_example.bar": "SELECT [MAP {'key1': 'value1'}, MAP {'key2': 'value2'}] AS array_col"
+    }
+
+    context.create_test(
```

Comment:
Can we reduce these three `create_test` calls into a single one by adding a couple more projections?

Reply:
Ditto about the maps tests

Reply:
Maybe one way to reduce the clutter is to have a single model we wanna create a test for. It would then contain as many projections as there are cases we want to test and we'd do it in one go to avoid the boilerplate. Wdyt?

Reply:
Isn't that what we're doing with the input_queries? We're reusing `sqlmesh_example.foo` and `sqlmesh_example.bar` and changing input_queries to the projections we want to test. There are a lot of unnecessary asserts that were copied and pasted I think we can remove.